jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1193,11 +1193,14 @@ public void write(boolean last, ByteBuffer content, Callback callback)
                 else
                 {
                     failure = getFailure(httpChannelState);
-                    if (failure == null && contentLength >= 0)
+                    if (failure == null && contentLength >= 0 && totalWritten != contentLength)
                     {
                         // If the content length were not compatible with what was written, then we need to abort.
-                        String lengthError = (totalWritten > contentLength) ? "written %d > %d content-length"
-                            : (last && totalWritten < contentLength) ? "written %d < %d content-length" : null;
+                        String lengthError = null;
```

Comment:
There doesn't seem to be any test coverage for when `lengthError != null`

Reply:
Thanks the normal path?!?!?

Reply:
That is the normal path and is explicitly tested by the HEAD requests with no body.